Library/ViewModels/DashboardViewModelTests.swift

```diff
@@ -335,4 +335,68 @@ internal final class DashboardViewModelTests: TestCase {
       self.vm.inputs.showHideProjectsDrawer()
     }
   }
+
+  func testTrackingEvents_CreatorDashboardViewed() {
+    let projects = (0...4).map { .template |> Project.lens.id .~ $0 }
+
+    withEnvironment(apiService: MockService(fetchProjectsResponse: projects)) {
+      XCTAssertEqual([], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      XCTAssertEqual(["Page Viewed"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillDisappear()
+
+      XCTAssertEqual(["Page Viewed"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      XCTAssertEqual(["Page Viewed", "Page Viewed"], self.segmentTrackingClient.events)
+    }
+  }
+
+  func testTrackingEvents_CreatorDashboardSwitchProjectClicked() {
+    let project1 = Project.template
+    let project2 = .template |> Project.lens.id .~ 4
+    let projects = [project1, project2]
+
+    withEnvironment(apiService: MockService(fetchProjectsResponse: projects)) {
+      XCTAssertEqual([], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
+
+      XCTAssertEqual(["Page Viewed"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.switch(toProject: .id(project2.id))
+
+      XCTAssertEqual(["Page Viewed", "CTA Clicked"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.switch(toProject: .id(project2.id))
```

Comment:
So you're probably not gonna switch from the same project to itself. So I guess if you start with project 1 you should switch project 2. Also along with the change to track the project being switched _to_, you can include a check to say the right project is being tracked when switching back and forth.

Reply:
So instead of using `[.template |>]` to check the assert values, you can pass in `project1` and `project2` directly. Makes the code less cumbersome, and easier to follow. Should pass tests, if it doesn't we'll revisit.